src/json-forms.ts

```diff
@@ -40,30 +42,39 @@ export class JsonForms extends HTMLElement {
     if (this.lastChild !== null) {
       this.removeChild(this.lastChild);
     }
-
-    if (this.services.length === 0) {
-      this.createServices();
-      this.dataService.initialRootRun();
+    if (this.services.length !== 0) {
+      this.services.forEach(service => service.dispose());
+      this.services = [];
     }
 
+    this.createServices();
+
     const bestRenderer = JsonFormsHolder.rendererService
         .getBestRenderer(this.uischema, this.dataschema, this.dataService);
     this.appendChild(bestRenderer);
+
+    this.dataService.initialRootRun();
   }
 
   set data(data: Object) {
     this.dataService = new DataService(data);
-    this.render();
+    if (this.allowDynamicUpdate) {
+      this.render();
+    }
   }
 
   set uiSchema(uischema: UISchemaElement) {
     this.uischema = uischema;
-    this.render();
+    if (this.allowDynamicUpdate) {
+      this.render();
+    }
   }
 
   set dataSchema(dataschema: JsonSchema) {
     this.dataschema = dataschema;
-    this.render();
+    if (this.allowDynamicUpdate) {
```

Comment:
Since the `render` call is guarded by `this.allowDynamicUpdate` so often, wouldn't it makes sense to move the check into the `render` method directly?